test/serializers/associations_test.rb

```diff
@@ -290,6 +289,111 @@ def test_illegal_conditional_associations
+          has_many :comments, custom_option: true, key: :reviews
+        end
+
+        InheritedAuthorSerializer = Class.new(AuthorSerializer) do
+          has_many :posts, custom_option: true
+          has_one :roles, custom_option: true, key: :new_roles
+          has_one :bio, custom_option: true
+        end
+
+        def setup
+          @author = Author.new(name: 'Steve K.')
+          @author.bio = Bio.new(id: 1, content: 'AMS Contributor')
+          @blog = Blog.new(name: 'AMS Blog')
+          @post = Post.new(title: 'New Post', body: 'Body')
+          @post.blog = @blog
+          @post.author = @author
+
+          @post_serializer = PostSerializer.new(@post, custom_options: true)
+          @author_serializer = AuthorSerializer.new(@author)
+          @inherited_post_serializer = InheritedPostSerializer.new(@post, custom_options: true)
+          @inherited_author_serializer = InheritedAuthorSerializer.new(@author)
+        end
+
+        def test_redefined_has_many_and_has_one
+          without_redefine = lambda do
+            associations = {}
+            @author_serializer.associations.each do |ass|
+              associations[ass.name] = ass.options[:custom_option]
+            end
+
+            expected_associations = {
+              posts: nil,
+              roles: nil,
+              bio: nil
+            }
+
+            assert_equal(associations, expected_associations)
+            assert_equal(@author_serializer.associations.count, 3)
+          end
+
+          without_redefine.call
+
+          with_redefine = lambda do
+            associations = {}
+            @inherited_author_serializer.associations.each do |ass|
+              associations[ass.name] = ass.options[:custom_option]
+            end
+
+            expected_associations = {
+              posts: true,
+              roles: true,
+              bio: true
+            }
+
+            assert_equal(associations, expected_associations)
+            # it should not replace association with a different key
+            assert_equal(@inherited_author_serializer.associations.count, 4)
+          end
+
+          with_redefine.call
+        end
+
+        def test_redefined_belongs_to
+          without_redefine = lambda do
+            associations = {}
+            @post_serializer.associations.each do |ass|
+              associations[ass.name] = ass.options[:custom_option]
+            end
+
+            expected_associations = {
+              comments: nil,
+              blog: nil,
+              author: nil
+            }
+
+            assert_equal(associations, expected_associations)
+            assert_equal(@author_serializer.associations.count, 3)
+          end
+
+          without_redefine.call
+
+          with_redefine = lambda do
+            associations = {}
+            @inherited_post_serializer.associations.each do |ass|
+              associations[ass.name] = ass.options[:custom_option]
```

Comment:
ah, so you're using custom option to test the presence of an association. 
This feels kinda hacky, and adds additional logic into tests that isn't part of the public API.

can you re-work these tests to use public API things? thanks!